### ISSUE TITLE
Syntax highlighting error for V language

### ISSUE THREAD

**Issue by MrTowers** [helix-editor/helix#6958]
### Summary

So whenever i try to type anything in string literal like: "hello", after one letter syntax highlighting completly brokes. I have language set to v

### Reproduction Steps

 

I tried this:

1. `hx`
2. open any v file
3. set language to v
4. type any string literal example: `"hello world"`
5. highlighting broken

I expected this to happen:
good highlighting
![image](https://user-images.githubusercontent.com/30564150/236069602-584e8b7a-9e9c-4205-a605-4aa9b48ae0af.png)

Instead, this happened:
broken highlighting
![image](https://user-images.githubusercontent.com/30564150/236069926-896a4213-8070-4252-86ab-c970bac6a126.png)

and what's interesting to fix this i just go to the `"` at the beginning of the string and insert space and remove and the highlighting just comes back to normal
![image](https://user-images.githubusercontent.com/30564150/236070869-01c49dc7-24ed-4fc6-85f4-6277263be8e3.png)

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2023-05-04T01:07:58.262 helix_term::application [DEBUG] received editor event: IdleTimer
2023-05-04T01:07:58.263 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 6
2023-05-04T01:07:58.293 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 6
2023-05-04T01:07:58.572 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 6
2023-05-04T01:07:58.752 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 6
2023-05-04T01:07:58.851 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 6
2023-05-04T01:07:59.112 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 6
2023-05-04T01:07:59.511 helix_term::application [DEBUG] received editor event: IdleTimer
2023-05-04T01:07:59.527 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 6
2023-05-04T01:07:59.530 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 6
2023-05-04T01:07:59.608 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 6
2023-05-04T01:07:59.761 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 6
2023-05-04T01:07:59.852 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 6
2023-05-04T01:08:00.254 helix_term::application [DEBUG] received editor event: IdleTimer
2023-05-04T01:08:00.254 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 6
2023-05-04T01:08:02.379 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 6
2023-05-04T01:08:02.565 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 6
2023-05-04T01:08:02.754 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 6
2023-05-04T01:08:02.882 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 6
2023-05-04T01:08:02.938 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 6
2023-05-04T01:08:03.138 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 6
2023-05-04T01:08:03.144 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 6
2023-05-04T01:08:03.484 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:08:03.491 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:08:03.771 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:08:03.907 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:08:04.308 helix_term::application [DEBUG] received editor event: IdleTimer
2023-05-04T01:09:16.320 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:09:16.605 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:09:17.006 helix_term::application [DEBUG] received editor event: IdleTimer
2023-05-04T01:09:17.420 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:09:17.617 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:09:17.892 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:09:18.294 helix_term::application [DEBUG] received editor event: IdleTimer
2023-05-04T01:09:18.314 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:09:18.509 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:09:18.780 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:09:18.962 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:09:19.156 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:09:19.558 helix_term::application [DEBUG] received editor event: IdleTimer
2023-05-04T01:10:43.285 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:10:43.686 helix_term::application [DEBUG] received editor event: IdleTimer
2023-05-04T01:10:43.793 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:10:43.823 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:10:43.856 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:10:43.886 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:10:43.918 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:10:43.949 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:10:43.981 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:10:44.011 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:10:44.047 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:10:44.076 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:10:44.109 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:10:44.137 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:10:44.361 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:10:44.527 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:10:44.928 helix_term::application [DEBUG] received editor event: IdleTimer
2023-05-04T01:10:45.114 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:10:45.237 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:10:45.639 helix_term::application [DEBUG] received editor event: IdleTimer
2023-05-04T01:10:45.639 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:10:45.804 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:10:46.125 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:10:46.131 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:10:46.526 helix_term::application [DEBUG] received editor event: IdleTimer
2023-05-04T01:11:09.221 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:11:09.681 helix_term::application [DEBUG] received editor event: IdleTimer
2023-05-04T01:11:10.059 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:11:10.185 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:11:10.319 helix_view::document [DEBUG] submitting save of doc 'Some("/home/shade/repo/v/naukacomp/src/main.v")'
2023-05-04T01:11:10.319 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 7
2023-05-04T01:11:10.330 helix_term::application [DEBUG] received editor event: DocumentSaved(Ok(DocumentSavedEvent { revision: 7, doc_id: DocumentId(2), path: "/home/shade/repo/v/naukacomp/src/main.v", text: ["module main\n\nfn main() {\n\t//after changing strign\n\tprintln(\"hello world! wow\");\n}"] }))
2023-05-04T01:11:10.330 helix_term::application [DEBUG] document Some("/home/shade/repo/v/naukacomp/src/main.v") saved with revision 7
2023-05-04T01:11:10.330 helix_view::document [DEBUG] doc 2 revision updated 1 -> 7
2023-05-04T01:11:10.330 helix_view::editor [DEBUG] editor status: 'src/main.v' written, 6L 81B
2023-05-04T01:11:10.331 helix_view::document [DEBUG] id 2 modified - last saved: 7, current: 7
2023-05-04T01:11:10.460 helix_term::application [DEBUG] received editor event: IdleTimer
2023-05-04T01:11:10.952 helix_view::document [DEBUG] id 2 modified - last saved: 7, current: 7
2023-05-04T01:11:11.353 helix_term::application [DEBUG] received editor event: IdleTimer
2023-05-04T01:11:17.051 helix_term::commands::typed [DEBUG] quitting...
2023-05-04T01:11:17.052 helix_view::document [DEBUG] id 2 modified - last saved: 7, current: 7
2023-05-04T01:11:17.159 helix_term::commands::typed [DEBUG] quitting...
2023-05-04T01:11:17.159 helix_view::document [DEBUG] id 2 modified - last saved: 7, current: 7
2023-05-04T01:11:17.159 helix_term::job [DEBUG] waiting on jobs...
2023-05-04T01:11:17.159 helix_term::job [DEBUG] waiting on jobs...
2023-05-04T01:11:51.806 helix_view::clipboard::provider [DEBUG] No native clipboard provider found. Yanking by OSC 52 and pasting will be internal to Helix
2023-05-04T01:11:51.812 helix_tui::backend::crossterm [DEBUG] The keyboard enhancement protocol is not supported in this terminal (checked in 724.7µs)
2023-05-04T01:11:51.812 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-05-04T01:11:52.209 helix_term::application [DEBUG] received editor event: IdleTimer
2023-05-04T01:11:53.396 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-05-04T01:11:53.477 globset [DEBUG] glob converted to regex: Glob { glob: "*.{zip,gz,bz2,zst,lzo,sz,tgz,tbz2,lz,lz4,lzma,lzo,z,Z,xz,7z,rar,cab}", re: "(?-u)^[^/]*\\.(cab|rar|7z|xz|Z|z|lzo|lzma|lz4|lz|tbz2|tgz|sz|lzo|zst|bz2|gz|zip)$", opts: GlobOptions { case_insensitive: false, literal_separator: true, backslash_escape: true }, tokens: Tokens([ZeroOrMore, Literal('.'), Alternates([Tokens([Literal('c'), Literal('a'), Literal('b')]), Tokens([Literal('r'), Literal('a'), Literal('r')]), Tokens([Literal('7'), Literal('z')]), Tokens([Literal('x'), Literal('z')]), Tokens([Literal('Z')]), Tokens([Literal('z')]), Tokens([Literal('l'), Literal('z'), Literal('o')]), Tokens([Literal('l'), Literal('z'), Literal('m'), Literal('a')]), Tokens([Literal('l'), Literal('z'), Literal('4')]), Tokens([Literal('l'), Literal('z')]), Tokens([Literal('t'), Literal('b'), Literal('z'), Literal('2')]), Tokens([Literal('t'), Literal('g'), Literal('z')]), Tokens([Literal('s'), Literal('z')]), Tokens([Literal('l'), Literal('z'), Literal('o')]), Tokens([Literal('z'), Literal('s'), Literal('t')]), Tokens([Literal('b'), Literal('z'), Literal('2')]), Tokens([Literal('g'), Literal('z')]), Tokens([Literal('z'), Literal('i'), Literal('p')])])]) }
2023-05-04T01:11:53.477 globset [DEBUG] built glob set; 0 literals, 0 basenames, 0 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 1 regexes
2023-05-04T01:11:53.480 globset [DEBUG] built glob set; 0 literals, 7 basenames, 8 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 0 regexes
2023-05-04T01:11:53.480 ignore::walk [DEBUG] ignoring /home/shade/repo/v/naukacomp/.editorconfig: Ignore(IgnoreMatch(Hidden))
2023-05-04T01:11:53.480 ignore::walk [DEBUG] ignoring /home/shade/repo/v/naukacomp/.git: Ignore(IgnoreMatch(Hidden))
2023-05-04T01:11:53.480 ignore::walk [DEBUG] ignoring /home/shade/repo/v/naukacomp/.gitattributes: Ignore(IgnoreMatch(Hidden))
2023-05-04T01:11:53.480 ignore::walk [DEBUG] ignoring /home/shade/repo/v/naukacomp/.gitignore: Ignore(IgnoreMatch(Hidden))
2023-05-04T01:11:53.480 helix_term::ui [DEBUG] file_picker init 6.5356ms
2023-05-04T01:11:53.481 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-05-04T01:11:53.875 helix_term::application [DEBUG] received editor event: IdleTimer
2023-05-04T01:11:53.901 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-05-04T01:11:54.219 helix_vcs [INFO] PeelToCommit(
    Unborn {
        name: FullName(
            "refs/heads/master",
        ),
    },
)
2023-05-04T01:11:54.219 helix_vcs [INFO] failed to open diff base for /home/shade/repo/v/naukacomp/src/main.v
2023-05-04T01:11:54.221 helix_vcs [INFO] PeelToCommit(
    Unborn {
        name: FullName(
            "refs/heads/master",
        ),
    },
)
2023-05-04T01:11:54.221 helix_vcs [INFO] failed to obtain current head name for /home/shade/repo/v/naukacomp/src/main.v
2023-05-04T01:11:54.225 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-05-04T01:11:54.225 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:54.225 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"initialize","params":{"capabilities":{"general":{"positionEncodings":["utf-8","utf-32","utf-16"]},"textDocument":{"codeAction":{"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["","quickfix","refactor","refactor.extract","refactor.inline","refactor.rewrite","source","source.organizeImports"]}}},"completion":{"completionItem":{"deprecatedSupport":true,"insertReplaceSupport":true,"resolveSupport":{"properties":["documentation","detail","additionalTextEdits"]},"snippetSupport":true,"tagSupport":{"valueSet":[1]}},"completionItemKind":{}},"hover":{"contentFormat":["markdown"]},"inlayHint":{"dynamicRegistration":false},"publishDiagnostics":{"versionSupport":true},"rename":{"dynamicRegistration":false,"honorsChangeAnnotations":false,"prepareSupport":true},"signatureHelp":{"signatureInformation":{"activeParameterSupport":true,"documentationFormat":["markdown"],"parameterInformation":{"labelOffsetSupport":true}}}},"window":{"workDoneProgress":true},"workspace":{"applyEdit":true,"configuration":true,"didChangeConfiguration":{"dynamicRegistration":false},"executeCommand":{"dynamicRegistration":false},"inlayHint":{"refreshSupport":false},"symbol":{"dynamicRegistration":false},"workspaceEdit":{"documentChanges":true,"failureHandling":"abort","normalizesLineEndings":false,"resourceOperations":["create","rename","delete"]},"workspaceFolders":true}},"clientInfo":{"name":"helix","version":"23.03 (4cdba7cc)"},"processId":918,"rootPath":"/home/shade/repo/v/naukacomp","rootUri":"file:///home/shade/repo/v/naukacomp","workspaceFolders":[{"name":"naukacomp","uri":"file:///home/shade/repo/v/naukacomp"}]},"id":0}
2023-05-04T01:11:54.236 helix_lsp::transport [ERROR] err <- "v ls error: Language server is not installed nor found.; code: 101 (101)\n"
2023-05-04T01:11:54.236 helix_lsp::transport [ERROR] err <- "\n"
2023-05-04T01:11:54.236 helix_lsp::transport [ERROR] err <- "- If you are using this for the first time, please run\n"
2023-05-04T01:11:54.236 helix_lsp::transport [ERROR] err <- "  `v ls --install` first to download and install VLS.\n"
2023-05-04T01:11:54.236 helix_lsp::transport [ERROR] err <- "- If you are using a custom version of VLS, check if\n"
2023-05-04T01:11:54.236 helix_lsp::transport [ERROR] err <- "  the specified path exists and is a valid executable.\n"
2023-05-04T01:11:54.236 helix_lsp::transport [ERROR] err <- "- If you have an existing installation of VLS, be sure\n"
2023-05-04T01:11:54.236 helix_lsp::transport [ERROR] err <- "  to remove \"vls.config.json\" and \"bin\" located inside\n"
2023-05-04T01:11:54.236 helix_lsp::transport [ERROR] err <- "  \"$HOME/.vls\" and re-install.\n"
2023-05-04T01:11:54.236 helix_lsp::transport [ERROR] err <- "\n"
2023-05-04T01:11:54.236 helix_lsp::transport [ERROR] err <- "  If none of the options listed have solved your issue,\n"
2023-05-04T01:11:54.236 helix_lsp::transport [ERROR] err <- "  please report it at https://github.com/vlang/v/issues\n"
2023-05-04T01:11:54.236 helix_lsp::transport [ERROR] err <- "\n"
2023-05-04T01:11:54.236 helix_lsp::transport [ERROR] err <- "/tmp/v_1000/vls.13975832662863874631.tmp.c:9136: at print_backtrace: Backtrace\n"
2023-05-04T01:11:54.236 helix_lsp::transport [ERROR] err <- "/tmp/v_1000/vls.13975832662863874631.tmp.c:20932: by main__VlsUpdater_cli_error\n"
2023-05-04T01:11:54.236 helix_lsp::transport [ERROR] err <- "/tmp/v_1000/vls.13975832662863874631.tmp.c:21029: by main__main\n"
2023-05-04T01:11:54.237 helix_lsp::transport [ERROR] err <- "/tmp/v_1000/vls.13975832662863874631.tmp.c:21513: by main\n"
2023-05-04T01:11:54.237 helix_lsp::transport [ERROR] err: <- StreamClosed
2023-05-04T01:11:54.237 helix_term::application [DEBUG] received editor event: LanguageServerMessage((0, Notification(Notification { jsonrpc: None, method: "exit", params: None })))
2023-05-04T01:11:54.237 helix_view::editor [DEBUG] editor status: Language server exited
2023-05-04T01:11:54.237 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:54.237 helix_lsp [ERROR] failed to initialize language server: server closed the stream
2023-05-04T01:11:54.616 helix_term::application [DEBUG] received editor event: IdleTimer
2023-05-04T01:11:54.756 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:54.991 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:55.181 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:55.395 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:55.552 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:55.938 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:56.087 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:56.489 helix_term::application [DEBUG] received editor event: IdleTimer
2023-05-04T01:11:56.610 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:56.639 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:56.668 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:56.701 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:56.731 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:56.761 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:56.791 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:56.823 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:56.854 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:56.884 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:56.916 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:56.949 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:56.978 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:57.011 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:57.041 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:57.072 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:57.103 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:57.133 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:57.166 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:57.194 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:57.227 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:57.258 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:57.287 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:57.320 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:57.351 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:57.558 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:57.740 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:58.039 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:58.440 helix_term::application [DEBUG] received editor event: IdleTimer
2023-05-04T01:11:58.546 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:58.686 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:58.794 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:58.940 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:59.102 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
2023-05-04T01:11:59.315 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 1
2023-05-04T01:11:59.322 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 1
2023-05-04T01:11:59.716 helix_term::application [DEBUG] received editor event: IdleTimer
2023-05-04T01:12:00.906 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 1
2023-05-04T01:12:01.308 helix_term::application [DEBUG] received editor event: IdleTimer
2023-05-04T01:12:02.455 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 1
2023-05-04T01:12:02.619 helix_view::document [DEBUG] submitting save of doc 'Some("/home/shade/repo/v/naukacomp/src/main.v")'
2023-05-04T01:12:02.619 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 1
2023-05-04T01:12:02.627 helix_term::application [DEBUG] received editor event: DocumentSaved(Ok(DocumentSavedEvent { revision: 1, doc_id: DocumentId(2), path: "/home/shade/repo/v/naukacomp/src/main.v", text: ["module main\n\nfn main() {\n\t//after changing strign\n\tprintln(\"hello world! wow lol\");\n}"] }))
2023-05-04T01:12:02.627 helix_term::application [DEBUG] document Some("/home/shade/repo/v/naukacomp/src/main.v") saved with revision 1
2023-05-04T01:12:02.627 helix_view::document [DEBUG] doc 2 revision updated 0 -> 1
2023-05-04T01:12:02.627 helix_view::editor [DEBUG] editor status: 'src/main.v' written, 6L 85B
2023-05-04T01:12:02.627 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 1
2023-05-04T01:12:03.250 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 1
2023-05-04T01:12:03.389 helix_term::commands::typed [DEBUG] quitting...
2023-05-04T01:12:03.389 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 1
2023-05-04T01:12:03.444 helix_term::commands::typed [DEBUG] quitting...
2023-05-04T01:12:03.444 helix_view::document [DEBUG] id 2 modified - last saved: 1, current: 1
2023-05-04T01:12:03.444 helix_term::job [DEBUG] waiting on jobs...
2023-05-04T01:12:03.444 helix_term::job [DEBUG] waiting on jobs...

```

</details>

### Platform

Windows 10

### Terminal Emulator

WSL

### Helix Version

23.03-127-gc6228825

**Comment by the-mikedavis** [comment]
This is most likely a bug upstream in tree-sitter-v's scanner (https://github.com/vlang/vls/blob/master/tree_sitter_v/src/scanner.c) which parses strings.

**Comment by oneslash** [comment]
@the-mikedavis, @MrTowers There is new LSP is released which replaced the old one, https://blog.vosca.dev/meet-v-analyzer/

I think we can close this ticket

**Comment by pascalkuthe** [comment]
in helix syntax highlighting is handeled by TS not the LS. However, in this case it appears that there is indeed a new grammar maintained in the same repo which seems to be newever. To fix the issue the build in grammar definition needs to be updated to point at the new grammar (and queries will potentially need updates)

**Comment by rmehri01** [comment]
Should this be closed by #7760?
